lib/matplotlib/pyplot.py

```diff
@@ -200,6 +200,126 @@ def uninstall_repl_displayhook():
 
 draw_all = _pylab_helpers.Gcf.draw_all
 
+_ENSURE_AX_DOC = """
+This function has been decorated by pyplot to have
+an implicit reference to the `plt.gca()` passed as the first argument.
+
+The wrapped function can be called as any of ::
+
+   func(*args, **kwargs)
+   func(ax, *args, **kwargs)
+   func(.., ax=ax)
+
+"""
+
+
+def ensure_ax(func):
+    """Decorator to ensure that the function gets an `Axes` object.
```

Comment:
If these catch on, that is probably worth it's own decorator.  

This mimics the expected behavior of `pyplot` (which is to use the current axes if it exists or create a new one).  Eventually I want to replace the implementation of a majority of pyplot with

```python
plot = ensure_ax(Axes.axes.plot)
hist = ensure_ax(Axes.axes.hist)
...
```